app/src/main/java/app/gamenative/data/DownloadInfo.kt

```
@@ -86,29 +142,91 @@ data class DownloadInfo(
+        if (force) {
+            val expectedGeneration = snapshotWriteGeneration.get()
+            try {
+                val persisted = persistResumeSnapshotInternal(
+                    appDirPath = appDirPath,
+                    expectedGeneration = expectedGeneration,
+                )
+                if (persisted) {
+                    lastPersistTimestampMs.set(nowMs)
+                    hasDirtyProgressSnapshot.set(false)
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)